x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/rest/VectorTileUtils.java

```diff
@@ -60,8 +60,10 @@ public static void addToXContentToFeature(VectorTile.Tile.Feature.Builder featur
      * Adds the provided key / value pair into the feature as tags.
      */
     public static void addPropertyToFeature(VectorTile.Tile.Feature.Builder feature, MvtLayerProps layerProps, String key, Object value) {
-        feature.addTags(layerProps.addKey(key));
-        feature.addTags(layerProps.addValue(value));
+        if (key != null && value != null) { // just make sure we don't try to add a null values
```

Comment:
I presume this defensive coding it not really required because the fix in RestVectorTileAction should already remove the nulls? Do we have this code in case there are other code paths we've not thought of that might cause nulls? Would it make more sense to throw an error here so we don't hide bugs?

Reply:
I removed the defensive coding. I added more test and it seems we are never passing null values here.